IDE0001: False Positive for default primary constructor

<details>
<summary>

#### Version used: `dotnet --info`

</summary>

```
.NET SDK:
 Version:           8.0.400
 Commit:            36fe6dda56
 Workload version:  8.0.400-manifests.56cd0383
 MSBuild version:   17.11.3+0c8610977

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.400\

Installierte .NET-Workloads:
Konfiguriert für die Verwendung loose manifests beim Installieren neuer Manifeste.
Es sind keine installierten Workloads zum Anzeigen vorhanden.

Host:
  Version:      8.0.8
  Architecture: x64
  Commit:       08338fcaa5

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.202 [C:\Program Files\dotnet\sdk]
  6.0.321 [C:\Program Files\dotnet\sdk]
  8.0.400 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

**Steps to Reproduce**:
Class, with default constructor:
```
// A
public abstract class Base<TOther>;
public sealed class Concrete() : Base<Concrete.Impl>() // IDE0001: Simplify Name
{
	public class Impl;
}

// B
public abstract class Base<TOther>();
public sealed class Concrete() : Base<Concrete.Impl>() // IDE0001: Simplify Name
{
	public class Impl;
}
```

Record class, with argument:
```
public abstract record class Base<TOther>(object A);
public sealed record class Concrete() : Base<Concrete.Impl>(default!) // IDE0001: Simplify Name
{
	public class Impl;
}
```

**Diagnostic Id**: `IDE0001: Simplify Name`

**Expected Behavior**: No false-positive, just like this does not produce a false-positive:
```
public abstract class Base<TOther>();

public sealed class Concrete() : Base<Concrete.Impl>
{
	public class Impl;
}
```

**Actual Behavior**: IDE0001 false-positive is emitted for `: Base<Concrete.Impl>()`, where it shouldn't.

I'm not sure what you mean by "Where the generic parameter E inside the base list is getting incorrectly bound even though members of D are not accessible in this context." This isn't happening from what I can tell. As an example, this test passes:

```cs
        [Fact]
        public void QuickTest()
        {
            var source = """
                public abstract class C<T>;

                public sealed class D() : C<E>()
                {
                    public class E;
                }
                """;

            var comp = CreateCompilation(source);
            comp.VerifyDiagnostics(
                // (3,29): error CS0246: The type or namespace name 'E' could not be found (are you missing a using directive or an assembly reference?)
                // public sealed class D() : C<E>()
                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "E").WithArguments("E").WithLocation(3, 29)
            );

            var tree = comp.SyntaxTrees[0];
            var model = comp.GetSemanticModel(tree);
            var classD = tree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>().Skip(1).First();
            var cBaseList = classD.BaseList.Types[0].Type;
            var eArg = ((GenericNameSyntax)cBaseList).TypeArgumentList.Arguments[0];
            var e = model.GetSymbolInfo(eArg).Symbol;
            Assert.Null(e);
        }
```

Could you please clarify what you're seeing?

--- a/src/Features/CSharpTest/SimplifyTypeNames/SimplifyTypeNamesTests.cs
+++ b/src/Features/CSharpTest/SimplifyTypeNames/SimplifyTypeNamesTests.cs
@@ -2251,6 +2251,34 @@ public partial class SimplifyTypeNamesTests : AbstractCSharpDiagnosticProviderBa
             """);
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75026")]
+    public async Task SimplifyUnmentionableTypeParameter3()
+    {
+        await TestMissingInRegularAndScriptAsync(
+            """
+            public class C<T>;
+
+            public class D : C<[|D.E|]>
+            {
+                public class E;
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75026")]
+    public async Task SimplifyUnmentionableTypeParameter3_PrimaryCtor()
+    {
+        await TestMissingInRegularAndScriptAsync(
+            """
+            public class C<T>;
+
+            public class D() : C<[|D.E|]>()
+            {
+                public class E;
+            }
+            """);
+    }
+
     [Fact]
     public async Task TestGlobalAlias()
     {

--- a/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs
+++ b/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs
@@ -72,10 +72,22 @@ internal class SpeculationAnalyzer : AbstractSpeculationAnalyzer<
     {
         Debug.Assert(expression != null);
 
-        var parentNodeToSpeculate = expression
-            .AncestorsAndSelf(ascendOutOfTrivia: false)
-            .Where(CanSpeculateOnNode)
-            .LastOrDefault();
+        SyntaxNode previousNode = null;
+        SyntaxNode parentNodeToSpeculate = null;
+        foreach (var node in expression.AncestorsAndSelf(ascendOutOfTrivia: false))
+        {
+            if (CanSpeculateOnNode(node))
+            {
+                // Only speculate on PrimaryConstructorBaseTypeSyntax if we are inside the argument list
+                if (node.Kind() is not SyntaxKind.PrimaryConstructorBaseType ||
+                    previousNode.Kind() is SyntaxKind.ArgumentList)
+                {
+                    parentNodeToSpeculate = node;
+                }
+            }
+
+            previousNode = node;
+        }
 
         return parentNodeToSpeculate ?? expression;
     }

